### ISSUE TITLE
Templates in message field of notifications do not work anymore

### ISSUE THREAD

**Issue by c-lega** [nielsfaber/alarmo#1037]
### Checklist

- [X] I am using the latest version of Alarmo (latest version can be found [here](https://github.com/nielsfaber/alarmo/releases/latest))
- [X] I checked for similar existing requests (both open and closed) before posting.

### Alarmo Version

v1.10.4

### HA Version

2024.10.0

### Bug description

I have many notifications triggered by all kind of alarm events. As there are many, I used a template to include the time it occurred in the message of the notification. That is not working anymore (not sure since when, but it's been at least a week or so - and I update almost instantly any update available).
Now I only can used a couple of wildcards and my templates do not work. BTW I have an Android device.

### Steps to reproduce

Create a new notification, set target to a mobile device, set message to any template (I sue: "{{now().strftime('%m/%d/%Y - %H:%M:%S')}}", and then press "TRY IT->"
You'll see the notification in the device contains the template itself instead of the expected output.

**Comment by hanshanderson** [comment]
I can confirm the same issue, starting with the update to 2024.10.0

**Comment by Al3xand3R360** [comment]
Same issue here since update to HA2024.10

**Comment by Gifford47** [comment]
Same issue on 2024.10.1.

**Comment by elbel86** [comment]
Same issue here on 2024.10.1

**Comment by nielsfaber** [comment]
At this point I have no idea what caused this functionality to break or how to fix it.
It seems that a change in the latest HA release introduced this bug, but there is no mentioning of this in the release notes of HA.
The templates are processed on the side of HA, Alarmo simply forwards the message text when executing the service.
I will have to dive deeper into the recent code changes in HA to see if I can find the cause. Unfortunately this may take some time.
If anyone can find some relevant information, please share.

**Comment by uvjim** [comment]
I don't believe this is affecting only Alarmo. I have templates set on the notify group I use; they have also stopped working. I remember reading a warning in the logs (can't find it now) about templates being deprecated. So not sure if that may have an affect. Templates seem to work if I use them in a normal automation with `notify` though.

**Comment by kevinsaucier** [comment]
Just chiming in here with the same issue.  From info on the post above, seems like HA expects the integration to process the Template and pass plain text to Notify.  Odd, though, as I can call the exact same service from the Developer window and it works as expected.  

_Alexa Media is a custom component. It needs to process the templates before calling the notify service. You need to report that to the custom component author.__


So, this works in Developer/Actions but doesn't work with 'Try It' from Alarmo.   Hmmmmm....

```
service: notify.notify_kevin
data:
  message: Alarm Armed to Home {{ states('sensor.date_time') }}
  title: Don't Open the Doors!
  data:
    ttl: 0
    channel: ha_info
    actions:
      - action: Disarm Alarmo
        title: Disarm Alarmo
      - action: Turn off Inside Lights
        title: Turn off Inside Lights
```

Also tried this, with the same results (thought maybe it had something to do with the change of Service to Action):

`action: notify.notify_kevin
data:
  message: Alarm Armed to Home {{ states('sensor.date_time') }}
  title: Don't Open the Doors!
  data:
    ttl: 0
    channel: ha_info
    actions:
      - action: Disarm Alarmo
        title: Disarm Alarmo
      - action: Turn off Inside Lights
        title: Turn off Inside Lights`

**Comment by thenextbutton** [comment]
Good Evening, apologies if this does not add anything to the conversation I was wondering what are the thoughts on this one? 

I have experienced the same when upgrading to 2024.10 and have rolled back 

I have the title for notifications set to {{ now().strftime('%H:%M %a %d %b %Y') }} on 2024.9 it fills in the information on 2024.10 I get the raw data.

Thanks in advance.

**Comment by Schermbecker** [comment]
Seems to be the same topic as seen here:
https://github.com/nielsfaber/alarmo/issues/989

**Comment by nielsfaber** [comment]
Just a small update: I am working on this issue, so far it looks fixable from the side of Alarmo.

**Comment by cmd85** [comment]
> Just a small update: I am working on this issue, so far it looks fixable from the side of Alarmo.

Thank you, was just about trying to start fixing workarounds in automations or rollback but I can wait if its being looked at, no real hurry =)

**Comment by nielsfaber** [comment]
After research I found out that support for templates in notifications is intentionally removed in the HA 2024.10 release, see  https://github.com/home-assistant/core/pull/122820.
The PR description states that templates can only be used in scripts/automations from now on, not in other integrations.
This was marked as breaking change but not mentioned in the release notes of HA.

I will keep digging and see if I can make some workaround, but I have to say I'm very surprised by this PR as well as the lack of communication of the change.

**Comment by Schermbecker** [comment]
@nielsfaber thank you so much for your effort! 
Alarmo is a great tool, I love it 😍

Alarmo provides properties "bypassed_sonsors" and "open_sensors" in its actions. I print this information into the appropriate notifications.

To do so I use template scripting just for an IF statement - to insert an additional label in case there are open or bypassed sensors.

Now (without the IF) the labels are always inserted, even if the properties are empty. So no big issue at all.

**Comment by hanshanderson** [comment]
I just updated to v1.10.6 and got following error message when I try to send a test notification:

`extra keys not allowed @ data['entity_id']. Got None`

Trying to save an existing notification leads to following error message:

`Something went wrong! Message format incorrect: string value is None for dictionary value @ data['actions'][0]['entity_id'] Response error: 400`

So I created a new notification with the same notification details and then it worked.:

```
service: notify.mobile_app_all
data:
  title: "{{ states('sensor.time') }}: Alarmo - Wohnung"
  message: |
    Alarmo: Der Alarm {{arm_mode|lang=de}} wurde ausgelöst!
    Ursache: {{open_sensors|lang=de}}.
  data:
    sticky: "true"
    tag: alarmo_wohnung_aktiv
    url: /dashboard-wohnung/alarm
    push:
      interruption-level: critical
```

**Comment by nielsfaber** [comment]
This issue should be resolved in the v1.10.6 release.

**Comment by c-lega** [comment]
@nielsfaber Thank you!... +1 for your effort!...

**Comment by thenextbutton** [comment]
Thank you so much for resolving this. Much appreciated

**Comment by KameDomotics** [comment]
> This issue should be resolved in the v1.10.6 release.

In the message, I would like to insert a template to show a text ONLY if there are no bypassed sensors.

```
message: >
  ALLARME ATTIVATO da {{ changed_by }} ({{ arm_mode | lang('it') }})
{% if bypassed_sensors and bypassed_sensors | length > 0 %} - Sensori esclusi: {{ bypassed_sensors }}{% else %} "" {% endif %}
```

Why doesn't it work?

**Comment by KameDomotics** [comment]
> > This issue should be resolved in the v1.10.6 release.
> 
> In the message, I would like to insert a template to show a text ONLY if there are no bypassed sensors.
> 
> ```
> message: >
>   ALLARME ATTIVATO da {{ changed_by }} ({{ arm_mode | lang('it') }})
> {% if bypassed_sensors and bypassed_sensors | length > 0 %} - Sensori esclusi: {{ bypassed_sensors }}{% else %} "" {% endif %}
> ```
> 
> Why doesn't it work?
> 


@nielsfaber 
Hi, is it possible to get this feature?

Thanks
